specs/NestedFrame.md

```diff
@@ -0,0 +1,268 @@
+    {
+        frame2->add_NavigationStarting(
+            Callback<ICoreWebView2FrameNavigationStartingEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    ICoreWebView2NavigationStartingEventArgs* args) -> HRESULT {
+                    std::wstring message = NavigationStartingArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::NavigationStarting");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+
+        frame2->add_ContentLoading(
+            Callback<ICoreWebView2FrameContentLoadingEventHandler>(
+                [this](ICoreWebView2Frame* sender, ICoreWebView2ContentLoadingEventArgs* args)
+                    -> HRESULT {
+                    std::wstring message = ContentLoadingArgsToJsonString(
+                        m_webviewEventSource.get(), args, L"CoreWebView2Frame::ContentLoading");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+        
+        frame2->add_DOMContentLoaded(
+            Callback<ICoreWebView2FrameDOMContentLoadedEventHandler>(
+                [this](ICoreWebView2Frame* sender, ICoreWebView2DOMContentLoadedEventArgs* args)
+                    -> HRESULT {
+                    std::wstring message = DOMContentLoadedArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::DOMContentLoaded");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+
+        frame2->add_NavigationCompleted(
+            Callback<ICoreWebView2FrameNavigationCompletedEventHandler>(
+                [this](
+                    ICoreWebView2Frame* sender,
+                    ICoreWebView2NavigationCompletedEventArgs* args) -> HRESULT {
+                    std::wstring message = NavigationCompletedArgsToJsonString(
+                        m_webviewEventSource.get(), args,
+                        L"CoreWebView2Frame::NavigationCompleted");
+                    PostEventMessage(message);
+                    return S_OK;
+                })
+                .Get(),
+            NULL);
+    }
+}
+
+void ScenarioWebViewEventMonitor::PostFrameCreatedEventMessage(wil::com_ptr<ICoreWebView2Frame> webviewFrame) {
+    wil::unique_cotaskmem_string name;
+    CHECK_FAILURE(webviewFrame->get_Name(&name));
+    auto frame5 = webviewFrame.try_query<ICoreWebView2Frame5>();
+    if (frame5)
+    {
+        UINT32 frameId = 0;
+        CHECK_FAILURE(frame5->get_FrameId(&frameId));
+    }
+
+    std::wstring message =
+        L"{ \"kind\": \"event\", \"name\": \"FrameCreated\", \"args\": {";
+    
+    message += L"\"frame\": " + EncodeQuote(name.get());

```

Comment:
Do we have a real-world sort of scenario where we would want to use this event? We want samples that demonstrates why an API is useful via a sample that an end dev could want to put in their app.